### PR TITLE
feat: image choice and repository

### LINKED ISSUE #565: Ability to choose image during build/deploy
### Is your feature request related to a problem? Please describe.
Now that Pepr's controller is in Iron Bank, we need the ability to choose images at build/deploy time in the cli 

### Describe the solution you'd like

- **Given** you are building or deploying using `npx pepr` and you want to use the IB image
- **When** you issue the command, there is a subcommand to decide which image to use
- **Then** the manifests contain the chosen image.

### Describe alternatives you've considered
(optional) A clear and concise description of any alternative solutions or features you've considered.

### PR DESCRIPTION
**Pull request by cmwylie19**
## Description

Add build command for custom image and registry to support Iron Bank

## Related Issue

Fixes #565 

Relates to #

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/pepr/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: docs/030_user-guide/010_pepr-cli.md [hunk 1]
```diff
@@ -86,4 +86,5 @@ Create a [zarf.yaml](https://zarf.dev) and K8s manifest for the current module.
 - `-o, --output-dir [output directory]` - Define where to place build output
 - `--timeout [timeout]` - How long the API server should wait for a webhook to respond before treating the call as a failure
 - `--rbac-mode [admin|scoped]` - Rbac Mode: admin, scoped (default: admin) (choices: "admin", "scoped", default: "admin")
-
+- `-ci, --custom-image [custom-image]` - Custom Image: Use custom image for Admission and Watcher Deployments.
```
Comment (btlghrants) [inline docs/030_user-guide/010_pepr-cli.md]:
```suggestion
- `-i, --custom-image [custom-image]` - Custom Image: Use custom image for Admission and Watcher Deployments.
```
Single char short flags.

https://github.com/tj/commander.js/blob/83c3f4e391754d2f80b179acc4bccc2d4d0c863d/lib/option.js#L310

File: src/cli/build.ts [hunk 2]
```diff
@@ -27,6 +27,10 @@ export default function (program: RootCmd) {
       "-n, --no-embed",
       "Disables embedding of deployment files into output module.  Useful when creating library modules intended solely for reuse/distribution via NPM.",
     )
+    .option(
+      "--custom-image [custom-image]",
+      "Custom Image: Use custom image for Admission and Watch Deployments.",
+    )
```
Comment (btlghrants) [inline src/cli/build.ts:33]: Short option..?

File: src/cli/build.ts [hunk 3]
```diff
@@ -27,6 +27,10 @@ export default function (program: RootCmd) {
       "-n, --no-embed",
       "Disables embedding of deployment files into output module.  Useful when creating library modules intended solely for reuse/distribution via NPM.",
     )
+    .option(
+      "--custom-image [custom-image]",
```
Comment (cmwylie19) [inline src/cli/build.ts]:
```suggestion
      "-i, --custom-image [custom-image]",
```

File: src/cli/build.ts [hunk 4]
```diff
@@ -95,6 +115,14 @@ export default function (program: RootCmd) {
         path,
       );
 
+      // If registry is set to Iron Bank, use Iron Bank image
+      if (opts.registry && opts.registry == "Iron Bank") {
```
Comment (cmwylie19) [inline src/cli/build.ts]:
```suggestion
      if (opts?.registry == "Iron Bank") {
```

**Review by btlghrants — APPROVED** [review]
'kay!

**Pull request merged**